opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -1135,19 +1139,39 @@ namespace detail {
             {
                 const ParallelISTLInformation& info =
                     boost::any_cast<const ParallelISTLInformation&>(pinfo);
-                int size_per_component = (end - it) / num_components;
+                typedef typename Iterator::value_type Scalar;
+                Scalar product = 0.0;
+                int size_per_component = (end - it);
+                size_per_component /= num_components; // two lines to supresse unused warning.
                 assert((end - it) == num_components * size_per_component);
-                double component_product = 0.0;
-                for( int i = 0; i < num_components; ++i )
+
+                if( num_components == 1 )
                 {
                     auto component_container =
-                        boost::make_iterator_range(it + i * size_per_component,
-                                                   it + (i + 1) * size_per_component);
+                        boost::make_iterator_range(it, end);
                     info.computeReduction(component_container,
                                            Opm::Reduction::makeInnerProductFunctor<double>(),
-                                           component_product);
+                                           product);
+                }
+                else
+                {
```

Comment:
Great catch! After comprehending your question, I don't understand that either. There is a reduction missing. Will fix it.